File: app/src/main/java/com/lexicalanalyzer/nusret/analyzer/Word.java
package com.lexicalanalyzer.nusret.analyzer;

/**
 * Created by Halil on 11.03.2018.
 */

public class Word {

    String word;
    String root;

    public Word(String word) {
        this.word = word;
        this.root = "undefined";
    }

    public Word(String word, String type) {
        this.word = word;
        this.root = type;
    }

    public String getWord() {
        return word;
    }

    public void setWord(String word) {
        this.word = word;
    }

    public String getType() {
        return root;
    }

    public void setType(String type) {
        this.root = type;
    }

    @Override
    public String toString() {

        if (!root.equals("undefined")) {
            return word + " " + root;
        } else
            return word;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Word word1 = (Word) o;

        return word.equals(word1.word);
    }

    @Override
    public int hashCode() {
        return word.hashCode();
    }
}

File: app/src/main/java/com/lexicalanalyzer/nusret/analyzer/Utils/Word.kt
package com.lexicalanalyzer.nusret.analyzer.Utils

/**
 * Created by Halil on 11.03.2018.
 */

class Word {

    var word: String
    var type: String

    constructor(word: String) {
        this.word = word
        this.type = "undefined"
    }

    constructor(word: String, type: String) {
        this.word = word
        this.type = type
    }

    override fun toString(): String {

        return if (type != "undefined") {
            "$word $type"
        } else
            word
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val word1 = o as Word?

        return word == word1!!.word
    }

    override fun hashCode(): Int {
        return word.hashCode()
    }
}
